src/include/wiredtiger.in

```diff
@@ -5654,13 +5655,55 @@ struct __wt_page_log {
     int (*terminate)(WT_PAGE_LOG *page_log, WT_SESSION *session);
 };
 
-/* Flags used with WT_PAGE_LOG_HANDLE::plh_put */
-/* AUTOMATIC FLAG VALUE GENERATION START 0 */
+/*!
+ * Values given to, or returned by the WT_PAGE_LOG_HANDLE::plh_put interface.
+ */
+struct __wt_page_log_put_args {
+       /*
+        * Input arguments
+        */
+       uint64_t backlink_checkpoint_id;
+       uint64_t base_checkpoint_id;
+
+       /* AUTOMATIC FLAG VALUE GENERATION START 0 */
 #define WT_PAGE_LOG_COMPRESSED 0x1u
 #define WT_PAGE_LOG_DELTA 0x2u
 #define WT_PAGE_LOG_ENCRYPTED 0x4u
-/* AUTOMATIC FLAG VALUE GENERATION STOP 32 */
+        /* AUTOMATIC FLAG VALUE GENERATION STOP 32 */
+       uint32_t flags;
+
+       /*
+        * Output arguments, returned by the call
+        */
+       uint64_t lsn;
+};
 
+/*!
+ * Values given to, or returned by the WT_PAGE_LOG_HANDLE::plh_get interface.
+ */
+struct __wt_page_log_get_args {
+       /*
+        * Input/output arguments
+        */
+       uint64_t lsn;    /* If non-zero, we are asking for a specific LSN */
+
+       /*
+        * Output arguments, returned by the call
+        */
+       uint64_t backlink_checkpoint_id;
+       uint64_t base_checkpoint_id;
+       uint32_t delta_count;
+};
+
+/*!
+ * The interface implemented by applications to provide handles used
+ * by a page log service implementation. This interface is returned by
+ * WT_PAGE_LOG::pl_open_handle .
+ *
+ * <b>Thread safety:</b> WiredTiger may invoke methods on the WT_PAGE_LOG_HANDLES
+ * interface from multiple threads concurrently. It is the responsibility of
+ * the implementation to protect any shared data.
```

Comment:
Should we also talk about reentrancy? I feel like it'd be a Bad Thing for the stuff on the other side of the PALI interface to somehow end up back in WiredTiger.